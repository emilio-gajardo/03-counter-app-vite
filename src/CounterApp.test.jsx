import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from './CounterApp';

describe('Test: <CounterApp />', () => {

    let initialValue = 1;
    test('1) Debe hacer match con el snapshot', () => {
        const { container } = render(<CounterApp value={initialValue} />);
        expect(container).toMatchSnapshot();
    });

    test(`2) Debe mostrar el valor inicial de <CounterApp value=${initialValue}>`, () => {
        render(<CounterApp value={initialValue} />);
        // expect(screen.getByText(value).innerHTML).toContain(`${initialValue}`);
        expect(screen.getByText(initialValue)).toBeTruthy();
    });

    test('3) Debe de incrementar con el botón [+1]', () => {
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('2')).toBeTruthy();
    });

    test('4) Debe de restar con el botón [-1]', () => {
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('0')).toBeTruthy();
    });

    test('5) Debe funcionar botón Reset', () => {
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
        expect(screen.getByText(1)).toBeTruthy();
    });
});