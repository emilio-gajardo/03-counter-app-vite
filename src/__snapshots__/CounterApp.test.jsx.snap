// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: <CounterApp /> 1) Debe hacer match con el snapshot 1`] = `
<div>
  <h2>
    CounterApp
  </h2>
  <h2>
    1
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
